test/compilable/reinterpretctfe.d

```diff
@@ -0,0 +1,10 @@
+// REQUIRED_ARGS:
+int nonPureFunc()
+{
+    return 2;
+}
+
+void main()
+{
+    static immutable ctfeI = (cast(int function() pure)&nonPureFunc);
```

Comment:
Mabye extend this test case to actually call the casted function pointer? (just to be sure)